com.unity.shadergraph/Editor/Data/Implementation/NodeUtils.cs

```diff
@@ -482,12 +482,16 @@ public static ShaderStageCapability GetEffectiveShaderStageCapability(MaterialSl
             var graph = initialSlot.owner.owner;
             s_SlotStack.Clear();
             s_SlotStack.Push(initialSlot);
+            ShaderStageCapability capabilities = ShaderStageCapability.All;
             while (s_SlotStack.Any())
             {
                 var slot = s_SlotStack.Pop();
-                ShaderStage stage;
-                if (slot.stageCapability.TryGetShaderStage(out stage))
```

Comment:
`TryGetShaderStage` was used to find if the capability was exactly one stage type. Turns out this isn't necessary with the right flag masking which would also handle new future stages.